### DISCUSSION TITLE
How can I track deep properties/objects change by using reaction?

### DISCUSSION THREAD

**Discussion by mannok** [mobxjs/mobx#3677] [Q&A]
By using [deepObserve](https://github.com/mobxjs/mobx-utils#deepobserve) we can track any nested properties/objects change. However, I don't find there is one for reaction? May I know if the can track deep properties/objects change by using reaction?

**Comment by mweststrate** [comment]
MobX tracks anything you access. So by merely reading the properties you
are interested in, it will start observing them. With autorun that happens
for the entire callback function, for react that happens for everything you
access in the first function you pass it.

**Reply by mannok** [reply]
@mweststrate thanks for help. However, for my case, I want to know if any of the object's nested property has been updated while the object has dynamic shape, by then I will trigger an alert to user. That means I am not accessing to any specific property of the object, what I need is whether there is any change under the the grand children of the objects

**Comment by mweststrate** [comment]
In that case access all the grandchildren

**Reply by mannok** [reply]
@mweststrate In my case the target object has dynamic data shape, so I am not able to know what grandchildren properties should be accessed

**Reply by mweststrate** [reply]
Sorry, I'm missing how that is relevant. You can loop over them like you'd do in any other place you'd be using them? Any normal JS approach to iterating whatever data structure you have should work, as long as the iteration isn't async.

**Reply by mweststrate** [reply]
e.g. `for (const key of obj) obj[key]` etc. etc. is all tracked

**Reply by mannok** [reply]
This could run into infinite circular dependency. Also, as I don't know the data shape of the target object, it can only be determined at runtime, it may contains nested object and nested property. The above solution assumed we know it at coding time and only 1 level of nested object

**Comment by mweststrate** [comment]
Reflecting on runtime is completely fine, how you traverse is up to you,
for MobX it doesn't matter, it will track with your traversal at runtime.
